Lesson4.1/src/com/home/service/Main.java

```diff
@@ -12,16 +12,15 @@ public static void main(String[] args) {
 
         int lifeCycles = computer.getLifeCycle();
 
-        while (lifeCycles > 0) {
-            int cycle = computer.turnOn(lifeCycles);
-            if (cycle > 0) {
-                lifeCycles--;
-            } else {
-                lifeCycles = 0;
+        while (lifeCycles >= 0) {
+            boolean isDead = false;
+            isDead = computer.turnOn(lifeCycles, isDead);
```

Comment:
instead of passing isDead to turn off or turn on just make it a global variable of the computer. It seems a bit unnatural that you are passing to computer information if it's dead or not. Computer should have this info